func.go

```diff
@@ -229,6 +240,21 @@ func RegisterFunc(fptr interface{}, cfn uintptr) {
 			runtime.KeepAlive(keepAlive)
 			runtime.KeepAlive(args)
 		}()
+		var syscall syscall15Args
+		if ty.NumOut() == 1 && ty.Out(0).Kind() == reflect.Struct {
+			outType := ty.Out(0)
+			if runtime.GOARCH == "amd64" && outType.Size() > 16 {
```

Comment:
Maybe check for `outType.Size() > 16` and then specialize for specific platform? Also it's not critical right now, as `darwin` -exclusivity is hardcoded up the code, but how about we panic we GOARCH is neither `amd64`, not `arm64`?

Reply:
No reason to add more checks in the actual function call since it's done on creation.